Lib9c/Model/Order/NonFungibleOrder.cs

```diff
@@ -109,6 +130,32 @@ public override OrderDigest Digest(AvatarState avatarState, CostumeStatSheet cos
                 $"Aborted because the tradable item({TradableId}) was failed to load from avatar's inventory.");
         }
 
+        public override OrderDigest Digest2(AvatarState avatarState, CostumeStatSheet costumeStatSheet)
+        {
+            if (avatarState.inventory.TryGetLockedItem(new OrderLock(OrderId), out Inventory.Item inventoryItem))
+            {
+                ItemBase item = inventoryItem.item;
+                int cp = CPHelper.GetCP((INonFungibleItem)item, costumeStatSheet);
+                int level = item is Equipment equipment ? equipment.level : 0;
+                return new OrderDigest(
+                    SellerAgentAddress,
+                    StartedBlockIndex,
+                    ExpiredBlockIndex,
+                    OrderId,
+                    TradableId,
+                    Price,
+                    cp,
+                    level,
+                    item.Id,
+                    1
+                );
+            }
+
+            throw new ItemDoesNotExistException(
+                $"Aborted because the tradable item({TradableId}) was failed to load from avatar's inventory.");
+        }
+
+        [Obsolete("Use ValidateCancelOrder2")]
```

Comment:
How about using the previous name(e.g. `ValidateCancelOrder`) as canonical and rename the previous method to others?